commons/all/src/main/java/org/infinispan/commons/api/query/EntityEntry.java

```diff
@@ -1,5 +1,5 @@
 package org.infinispan.commons.api.query;
 
-public record EntityEntry<K, E>(K key, E value, float score) {
+public record EntityEntry<K, E>(K key, E value, float score, Object metadata) {
```

Comment:
Why `Object metadata` and not `Metadata metadata`?

Reply:
`EntityEntry` is a common class, while `Metadata` is a core class. So I think that the latter is not visible from the former.

Reply:
That's a shame :cry: